### PR TITLE
Update and fix deps

### PR DESCRIPTION
**Pull request by SUPERCILEX**
@samtstern See https://github.com/firebase/FirebaseUI-Android/pull/752#issuecomment-308628100

### CODE REVIEW

File: library/google-services.json [hunk 1]
```diff
@@ -1,33 +1,25 @@
 {
   "project_info": {
-    "project_number": "",
-    "firebase_url": "",
-    "project_id": "",
-    "storage_bucket": ""
+    "project_number": "316052285104",
```
Comment (SUPERCILEX) [inline library/google-services.json]: `com.google.gms:google-services:3.1.0` complained with these empty values so I just provided my dummy project ones.

Comment (samtstern) [inline library/google-services.json]: Can we just put some actual dummy values then?  I know there's nothing bad about leaking yours but it feels a little odd...

**Comment by samtstern** [comment]
@SUPERCILEX thank you for solving this!  Much better, just left one comment.

**Comment by SUPERCILEX** [comment]
@samtstern Yup, rebased.

File: library/google-services.json (hunk 1 above)
Comment (SUPERCILEX) [inline library/google-services.json]: Yup, rebased.

**Comment by samtstern** [comment]
@SUPERCILEX d'oh looks like the plugin can detect obviously fake values.

Here's a fake JSON file I've made that works:
https://github.com/firebase/quickstart-android/blob/master/mock-google-services.json

You can copy the values you need out of that.

**Comment by SUPERCILEX** [comment]
@samtstern Dang, that's a hardcore `google-services.json` you have there!😆 Thanks!

**Pull request merged**